docs: add a how-to to build a basic image (#114)
canonical/imagecraft: docs/how-to-guides/basic_image.rst

> .. _howto_buld_basic_image:

Reviewer: ```suggestion
.. _how-to_build_basic_image:
```

===================
Build a basic image
===================

These instructions describe how to build a ``classic`` server 24.04 AMD64 Ubuntu image
with Imagecraft.

Prerequisites
-------------

- AMD64 machine with Ubuntu 24.04

> - ``snapd`` installed

Author: I opted for a link, because I would rather be very explicit on the prerequisites even if you are right that snapd will very likely be installed on a 24.04 machine.

- 15GiB or more disk space to process the build and hold the resulting image

.. note:: Following instructions build an **AMD64** image on an **AMD64** machine.
          Building on another architecture would need several modifications not
          described on this page.

Install Imagecraft
~~~~~~~~~~~~~~~~~~

Imagecraft is only available as a snap in the ``latest/edge`` channel

> `from the Snapstore <https://snapcraft.io/imagecraft>`_. Install it with:

Reviewer 2: ```suggestion
in the `Snap Store <https://snapcraft.io/imagecraft>`_. Install it with:
```

.. code-block::

>     sudo snap install --classic --edge imagecraft

Reviewer 2: ```suggestion
    sudo snap install imagecraft --channel=edge --classic
```
A minor difference, but this pattern more closely aligns with what guidance in Snacpraft.

> Verify that ``imagecraft`` is properly installed:

Reviewer 2: ```suggestion
Verify that Imagecraft is properly installed:
```
I'll have a rule for this in a guide in the next cycle, but essentially, when we're referring to a product, it should always be written in the plain-language format.

.. code-block::

    imagecraft

.. important:: ``imagecraft`` requires **elevated permissions**. Run it with **root**
               privileges or using ``sudo`` if you plan on using the executing system

>                as the build system ("destructive mode").

Reviewer 2: I'm having a hard time understanding the nuances here. I think we should make some changes to this callout.

- If Imagecraft _requires_ elevated permissions, why do we say "if you plan on using..."? The second phrase makes it sound optional.
- Since we're warning about destructive mode here, what are the potential side effects? It sounds dangerous – can we describe how?
- Maybe this callout should be a caution, not a warning? We use cautions for when a user action could damage data or hardware.

Author: I replaced this with a caution callout warning about the risk of using the tool. For now the rest of the how-to only mention running the tool with `sudo` to pack an image so I will revisit that when I am sure the tool can work well in lxd/multipass.

Reviewer 2: Looks good, thanks!

Prepare the configuration
-------------------------

Imagecraft configuration

> ~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer 2: Is this heading needed? It looks like we could cut it.

Save the following content as ``imagecraft.yaml``:

> .. literalinclude:: code/basic_imagecraft.yaml

Reviewer 2: Agreed – it's too long, unfortunately.

@upils We have examples of wrapped code in Snapcraft: https://github.com/canonical/snapcraft/blob/f4d43dcdb054514274721e862274242d1f424c6d/docs/howto/craft-a-snap/example-python-app.rst?plain=1#L22

    :caption: imagecraft.yaml
    :language: yaml

> .. note:: The name of the configuration file, ``imagecraft.yaml``, is **important**

Reviewer 2: Instead of making this a note after the reader copies the code, what do you think about making it a regular paragraph at the start of the section? I think it would be better if the user got this information before they took action.

          because Imagecraft uses it automatically.

Build the image
---------------

Build the image with destructive mode. In this case the machine must be of the series
of the ``build-base`` declared in the ``imagecraft.yaml`` file.

>   .. code-block::

Reviewer 2: ```suggestion
.. code-block::
```

The indentation is turning this into a block quote.

    sudo imagecraft --verbosity debug pack --destructive-mode

The resulting image file, ``pc.img``, is available in the current directory.

.. note:: Without any specific option imagecraft will rely by default on ``LXD``
          to build the image. However this mode of operation is not working yet.

Run the image
--------------

> Test the resulting image with QEMU.
> 
> Copy UEFI variables to a temporary directory:

Reviewer 2: ```suggestion
Finally, test your new image with QEMU.

First, copy the UEFI variables to a temporary directory:
```
